airflow/providers/amazon/aws/operators/emr.py

```diff
@@ -1025,11 +1034,12 @@ def __init__(
         self.aws_conn_id = aws_conn_id
         self.release_label = release_label
         self.job_type = job_type
-        self.wait_for_completion = wait_for_completion
+        self.wait_for_completion = False if deferrable else wait_for_completion
```

Comment:
> I was actually wondering with the new default if it'd make sense to defer only if deferrable && wait_for_completion ?

Why would that be the case? In this case `deferrable` and `wait_for_completion` don't really interact (i.e. if we defer, then we never check for `wait_for_completion`) But I think `deferrable=True` implies `wait_for_completion=True`.

A lot of the time, `wait_for_completion` is set as `False` by default, which would be confusing for users if they pass `deferrable=True` but we don't defer because `wait_for_completion` is `False`.

Reply:
what I'm saying is that if you have an operator where `wait_for_completion = False` and you set your config to `operators.default_deferrable = True` because yeah sure you want to free workers, then suddenly your operators start waiting when they weren't before and that can be a bit confusing

Reply:
the "good" situation I'd expect as a user by setting `operators.default_deferrable = True` is that my operators waiting for completion would start deferring, and those not waiting would continue to not wait

Reply:
You have a good point, but a counter point is that it would be confusing for users to set `deferrable=True` for an operator, and not see any difference in behaviour. 

> then suddenly your operators start waiting when they weren't before

I think the solution to this is to be very clear that `deferrable=True` implies `wait_for_completion=True`.
I think this is a decision that we need to clear up, and then implement everywhere. A quick search showed that we have an inconsistent definition of how `deferrable` and `wait_for_completion` work with each other.

Reply:
This question has been posted on the dev list, and we can continue it there. For now, I will resolve this conversation so that this PR does not get held up. Whatever decision we make, we will have to apply it to other operators anyways